### PR TITLE
feat(developer): ldml: more improvement for key-not-found 🙀 

### PR DESCRIPTION
**Pull request by srl295**
- add keys on each row and add rows, mapping to a new reserved "gap (reserved)" key.
- add a mechanism for reserved keys, in case we have others

For #9451

- [x] TODO: fix basic.txt

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc-ldml/src/compiler/keys.ts [hunk 1]
```diff
@@ -9,6 +9,11 @@ import ListItem = KMXPlus.ListItem;
 import KeysFlicks = KMXPlus.KeysFlicks;
 import { allUsedKeyIdsInFlick, allUsedKeyIdsInKey, allUsedKeyIdsInLayers, calculateUniqueKeys, hashFlicks, hashKeys, translateLayerAttrToModifier, validModifier } from '../util/util.js';
 import { MarkerTracker, MarkerUse } from './marker-tracker.js';
+import { KeysKeys } from '../../../../../common/web/types/src/kmx/kmx-plus.js';
```
Comment (mcdurdin) [inline developer/src/kmc-ldml/src/compiler/keys.ts:12]: This needs to be an import from `@keymanapp/common-types`

**Comment by srl295** [comment]
```
../common/web/types/build/src/kmx/string-list.d.ts(1,35): error TS2307: Cannot find module 'src/ldml-keyboard/pattern-parser.js' or its corresponding type declarations.
[18:14:07 ](https://build.palaso.org/buildConfiguration/Keymanweb_TestPullRequests/431218?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&hideProblemsFromDependencies=false&expandBuildProblemsSection=true&showLog=431218_1351_1351&logView=flowAware)  ../common/web/types/build/src/osk/osk.d.ts(1,33): error TS2307: Cannot find module 'src/keyman-touch-layout/keyman-touch-layout-file.js' or its corresponding type declarations.
```

strange errs

**Comment by mcdurdin** [comment]
Src/ is inserted by vscode but needs to be a ./ or ../ relative path

**Comment by srl295** [comment]
> Src/ is inserted by vscode but needs to be a ./ or ../ relative path

5 month old commit… it's failing in master, opened PR https://github.com/keymanapp/keyman/pull/10279

**Review by jahorton — APPROVED** [review]

**Comment by jahorton** [comment]
Web build-check failure will likely be resolved by #10279.

**Comment by srl295** [comment]
download failure would be mitigated by #9534

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.235-alpha](https://keyman.com/downloads/releases/alpha/17.0.235)

**Review by mcdurdin — COMMENTED** [review]
post-facto lgtm

**Pull request merged**